extendr-api/src/robj/mod.rs

```diff
@@ -352,6 +352,9 @@ impl Robj {
                 let sexp = self.get();
                 use SEXPTYPE::*;
                 match self.sexptype() {
+                    // a character vector contains `CHARSXP`, and thus you
+                    // seldomly have `Robj`'s that are `CHARSXP` themselves
+                    CHARSXP => sexp == libR_sys::R_NaString,
```

Comment:
If `match` branches precedence is important and `CHARSXP` is rare, move it to the end of the clause?